admin/post-types/writepanels/writepanel-certificate_data.php

```diff
@@ -68,248 +68,76 @@ function certificate_template_data_meta_box( $post ) {
-							'value'       => isset( $woothemes_sensei_certificate_templates->certificate_template_fields['certificate_completion']['text'] ) ? $woothemes_sensei_certificate_templates->certificate_template_fields['certificate_completion']['text'] : '',
-						)
-					);
-				echo '</div>';
 
-				// Certificate place position.
-				echo '<div class="options_group">';
-					certificate_templates_wp_position_picker(
-						array(
-							'id'          => 'certificate_place_pos',
-							'label'       => __( 'Place Position', 'sensei-certificates' ),
-							'value'       => implode( ',', $woothemes_sensei_certificate_templates->get_field_position( 'certificate_place' ) ),
-							'description' => __( 'Optional position of the place of Certification.', 'sensei-certificates' ),
-						)
-					);
-					certificates_wp_hidden_input(
-						array(
-							'id'    => '_certificate_place_pos',
-							'class' => 'field_pos',
-							'value' => implode( ',', $woothemes_sensei_certificate_templates->get_field_position( 'certificate_place' ) ),
-						)
-					);
-					certificate_templates_wp_font_select(
-						array(
-							'id'      => '_certificate_place',
-							'label'   => __( 'Font', 'sensei-certificates' ),
-							'options' => $available_fonts,
-						)
-					);
-					certificates_wp_text_input(
-						array(
-							'id'    => '_certificate_place_font_color',
-							'label' => __( 'Font color', 'sensei-certificates' ),
-							'value' => isset( $woothemes_sensei_certificate_templates->certificate_template_fields['certificate_place']['font']['color'] ) ? $woothemes_sensei_certificate_templates->certificate_template_fields['certificate_place']['font']['color'] : '',
-							'class' => 'colorpick',
-						)
-					);
-					certificates_wp_text_input(
+					$text_function = ( 'textarea' === $field_info['type'] ) ? 'certificates_wp_textarea_input' : 'certificates_wp_text_input';
+					$text_function(
 						array(
 							'class'       => 'medium',
-							'id'          => '_certificate_place_text',
-							'label'       => __( 'Course Place Text', 'sensei-certificates' ),
-							'description' => __( 'Text to display in the course place area.', 'sensei-certificates' ),
-							'placeholder' => __( '{{course_place}}', 'sensei-certificates' ),
-							'value'       => isset( $woothemes_sensei_certificate_templates->certificate_template_fields['certificate_place']['text'] ) ? $woothemes_sensei_certificate_templates->certificate_template_fields['certificate_place']['text'] : '',
+							'id'          => "_certificate_{$field_key}_text",
+							'label'       => $field_info['text_label'],
+							'description' => $field_info['text_description'],
+							'placeholder' => $field_info['text_placeholder'],
+							'value'       => isset( $woothemes_sensei_certificate_templates->certificate_template_fields[ "certificate_$field_key" ]['text'] ) ? $woothemes_sensei_certificate_templates->certificate_template_fields[ "certificate_$field_key" ]['text'] : '',
```

Comment:
```suggestion
							'value'       => isset( $woothemes_sensei_certificate_templates->certificate_template_fields[ "certificate_{$field_key}" ]['text'] ) ? $woothemes_sensei_certificate_templates->certificate_template_fields[ "certificate_{$field_key}" ]['text'] : '',
```

Reply:
I've addressed the above comments in https://github.com/woocommerce/sensei-certificates/pull/292/commits/92122b29b5377a90fdb7f9ffabb1f3f1583c4129.